contracts/svm/testing/tests/swap.rs

```diff
@@ -0,0 +1,568 @@
+        program_pack::Pack,
+        pubkey::Pubkey,
+        signature::{
+            Keypair,
+            Signer,
+        },
+        system_instruction,
+    },
+    testing::{
+        express_relay::{
+            helpers::{
+                get_express_relay_metadata,
+                get_express_relay_metadata_key,
+            },
+            set_swap_platform_fee::set_swap_platform_fee_instruction,
+            swap::build_swap_instructions,
+        },
+        helpers::{
+            assert_custom_error,
+            generate_and_fund_key,
+            submit_transaction,
+        },
+        setup::{
+            setup,
+            SetupResult,
+        },
+    },
+};
+
+pub struct Token {
+    pub mint:          Pubkey,
+    pub decimals:      u8,
+    mint_authority:    Keypair,
+    pub token_program: Pubkey,
+}
+
+impl Token {
+    pub fn airdrop(&self, svm: &mut LiteSVM, destination: &Pubkey) {
+        let instructions = vec![
+            create_associated_token_account_idempotent(
+                &self.mint_authority.pubkey(),
+                destination,
+                &self.mint,
+                &self.token_program,
+            ),
+            mint_to_checked(
+                &self.token_program,
+                &self.mint,
+                &get_associated_token_address_with_program_id(
+                    destination,
+                    &self.mint,
+                    &self.token_program,
+                ),
+                &self.mint_authority.pubkey(),
+                &[&self.mint_authority.pubkey()],
+                self.get_amount_with_decimals(10f64),
+                self.decimals,
+            )
+            .unwrap(),
+        ];
+        submit_transaction(
+            svm,
+            &instructions,
+            &self.mint_authority,
+            &[&self.mint_authority],
+        )
+        .unwrap();
+    }
+
+    pub fn token_balance_matches(svm: &mut LiteSVM, account: &Pubkey, amount: u64) -> bool {
+        let token_account_option = &mut svm.get_account(account).map(|account| {
+            anchor_spl::token_interface::TokenAccount::try_deserialize(&mut account.data.as_slice())
+                .unwrap()
+        });
+
+        if token_account_option.is_none() {
+            return amount == 0;
+        }
+
+        token_account_option.unwrap().amount == amount
+    }
+
+    pub fn create_token_account(&self, svm: &mut LiteSVM, owner: &Pubkey) -> Pubkey {
+        let token_account = Keypair::new();
+        let token_account_rent =
+            svm.minimum_balance_for_rent_exemption(spl_token_2022::state::Account::LEN);
+        let instructions = vec![
+            system_instruction::create_account(
+                &self.mint_authority.pubkey(),
+                &token_account.pubkey(),
+                token_account_rent,
+                spl_token_2022::state::Account::LEN.try_into().unwrap(),
+                &self.token_program,
+            ),
+            initialize_account2(
```

Comment:
por que los dos

Reply:
fixed

Reply:
nice. out of curiosity, what is the difference btwn 1 and 2 and 3?